### PR TITLE
Experiment with new `status()` display

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/status.R
Comment (hadley) [inline R/status.R:114]: I wonder if these should be errors.

File: R/status.R
Comment (hadley) [inline R/status.R:246]: Probably worth pulling this out into a separate function.

File: NEWS.md
Comment (kevinushey) [inline NEWS.md:5]:
```suggestion
  inconsistent combination of being installed, used, and recorded.
```

File: R/status.R
Comment (kevinushey) [inline R/status.R:9]:
```suggestion
#' of successfully `restore()`ing the project in the future or on another
```

File: R/status.R
Comment (kevinushey) [inline R/status.R:16]:
```suggestion
#' important to do first because if any packages are missing we can’t tell for
```

Comment (kevinushey) [reply R/status.R:16]:
```suggestion
#' important do to first because if any packages are missing we can't tell for
```

File: R/status.R
Comment (kevinushey) [inline R/status.R:17]:
```suggestion
#' sure that a package isn't used; it might be a dependency that we don’t know
```

File: R/status.R
Comment (kevinushey) [reply R/status.R:114]: I don't think so; IMHO `status()` shouldn't ever give an error, it should just give advice for getting the project into a "good" state.

File: R/status.R
Comment (kevinushey) [inline R/status.R:141]: This might no longer be necessary; we now implicitly add BiocManager and BiocVersion as dependencies to any Bioconductor packages we find.

Comment (kevinushey) [reply R/status.R:141]: https://github.com/rstudio/renv/blob/main/R/dependencies.R#L556-L562

File: R/status.R
Comment (kevinushey) [inline R/status.R:174]: For some reason I don't like the word 'problems' in this context: maybe instead we can use something like "reconciling this state" or something similar? That is, I don't view the lockfile + library being out of sync as a "problem" but just a thing that happens over the lifetime of a project that needs your decision-making.

File: R/status.R
Comment (hadley) [reply R/status.R:114]: My reasoning is that it might make sense for it to error when called on something that's not an renv project; you can obviously fix that by calling `renv::init()` but it might be more likely to reflect a genuine error? i.e. you're in the wrong directory.

Comment (kevinushey) [reply R/status.R:114]: Ah, that's a good point...

What if the user (1) hasn't loaded renv but (2) the working directory is in an renv project? (What if it's a sub-directory of an renv project?)

File: R/status.R
Comment (hadley) [reply R/status.R:141]: Yup, confirmed that it's not needed any more (the test we added for this case still passes).

File: R/status.R
Comment (hadley) [reply R/status.R:116]: What if we error if there's neither a project library nor a lockfile, but otherwise give a helpful message (i.e. call `snapshot()` if no lockfile; call `restore()` if no library)?

File: R/status.R
Comment (kevinushey) [inline R/status.R:238]:
```suggestion
    writef("The following package(s) are in an inconsistent state:")
```

File: NEWS.md
Comment (kevinushey) [inline NEWS.md:4]:
```suggestion
* `renv::status()` gets new, more compact, display when packages have some 
```

I try to stick to using `renv::` as a prefix because we often encourage users to prefix usages of `renv` in a similar way; e.g. to avoid confusion about `base::load()` versus `renv::load()`.

File: R/status.R
Comment (kevinushey) [inline R/status.R:21]: What should users do when multiple packages are in an inconsistent state? E.g. you have some packages which are used and recorded; and others which are recorded but not used.

(I think we want to recommend using `renv::restore()` first; perhaps `renv::restore(packages = <...>)`, and then using `renv::snapshot()` to remove unused packages?)

File: R/status.R
Comment (kevinushey) [inline R/status.R:62]: We might also recommend using `git` + `renv::history()` or `renv::revert()` for finding and using historical versions of lockfiles.

File: R/status.R
Comment (kevinushey) [reply R/status.R:116]: I think that makes sense.

**Review by kevinushey — APPROVED** [review]

File: R/status.R
Comment (kevinushey) [reply R/status.R:238]: Maybe pedantic but the package itself isn't inconsistent; it's the way it appears to be used / recorded in the project that is?

File: R/status.R
Comment (hadley) [reply R/status.R:116]: I've left these all as messages, not errors, but tweaked them all to be more immediately actionable.

**Comment by kevinushey** [comment]
LGTM; I'll let you merge when ready!

**Pull request merged**
